src/components/UserSearch.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react';
+import {Route} from 'react-router-dom';
+
+class UserSearch extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      githubUsername: '',
+      selectedUsername: '',
+    };
+
+    this.searchUser = this.searchUser.bind(this);
+    this.updateUsername = this.updateUsername.bind(this);
+  }
+
+  searchUser(evt) {
+    evt.preventDefault();
+    // TODO: Dispatch action that will request data from githubs API
+    console.log("Searching user....");
+    this.setState({selectedUsername: this.state.githubUsername});
+  }
+
+  updateUsername(evt) {
+    console.log(evt);
+    this.setState({
+      githubUsername: evt.target.value,
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.searchUser}>
```

Comment:
Instead of submitting the form and triggering a page refresh, why not make this an `onClick={this.searchUser}` on the `button` below?

Reply:
Yes I can make this change.